client/ayon_houdini/api/plugin.py

```diff
@@ -325,7 +326,27 @@ def filepath_from_context(cls, context):
         if cls.use_ayon_entity_uri:
             return get_ayon_entity_uri_from_representation_context(context)
 
-        return super(HoudiniLoader, cls).filepath_from_context(context)
+        path = super().filepath_from_context(context)
+
+        # Remap project roots to the relevant environment variables
+        # TODO: Refactor this to more optimal/better code
+        mapping = {}
+        for key, value in os.environ.items():
+            if key.startswith("AYON_PROJECT_ROOT_"):
+                mapping[key] = value
```

Comment:
I'm not sure if this is faster, but we may not need to go through the env keys because the anatomy object already has this piece of information. 
I wonder if there's a cached anatomy object instead of creating/querying it. 
```suggestion
        anatomy = Anatomy(context["project"]["name"], project_entity=context["project"])
        mapping = anatomy.root_environments()
```

Reply:
I'm thinking it may be better to just 'cache it once' in the current session since that part of `os.environ` (for the roots) shouldn't update on context changes regardless. Right @iLLiCiTiT ?

Reply:
> I mostly refrained from doing this due to speed reasons - getting the Anatomy object usually results in quite some queries which I hoped to avoid.

If you pass if `project_entity` it does not do any additional queries.

But what if you're loading from library project? The environments are completelly different in that case, or?

Reply:
Does having the key in os.environ mean it's actually added to hscript or doesn't it?
Honestly, I don't know because the last item I plated with env vars I was checking if the variables is set in both environ and hscript. https://github.com/ynput/ayon-houdini/blob/26f8cfb9d7f1b0a3a651d89161a825e8617a7d91/client/ayon_houdini/api/lib.py#L1013

Reply:
It is not set to `hscript` no, however houdini vars still expand them.

Houdini expands them in order:
- is hscript var? -> expand
- is env var? -> expand

Reply:
BTW with Anatomy you could just use this and skip `filepath_from_context`.
```python
env_roots = anatomy.roots.root_environmets_fill_data()
return get_representation_path(representation, env_roots)
```

Reply:
Does that _leave_ the roots in tact? Also, good point on only supporting _current project_ there!

> The envs were aded in Pype 2. We didn't have real usage for them since OpenPype, but didn't remove them.

If we do remove them - then it makes sense to explicitly 'enable' them in `ayon-houdini` to support this Houdini feature. We can make a PR for that once we need at that point in time.

Reply:
> Does that leave the roots in tact? Also, good point on only supporting current project there!

Default behavior of `root_environmets_fill_data` is to return env keys as values for root `{"work": "$AYON_AYON_PROJECT_ROOT_WORK", ...}` -> which is what you're guessing here, whereas the suggested code does fill it with the root that is really requested by the template.

Reply:
Ended up sticking closer to what I already had - just felt more reliable trusting on values that Houdini would actually currently detect as valid keys - no matter what the anatomy may say currently.